Add prose / example for headers attribute

scottaohara opened the pull request:
fixes #1359

Updates the `td` prose to mention of how the `headers` attribute can be used to explicitly indicate headers for a particular cell.

Revise note to refer to tables example to indicate that the `td`’s content attributes (`colspan`, `rowspan`, `headers`) will have examples in one of/both of the linked sections.

Add table with `headers` example to table examples at end of file.

scottaohara committed:
fixes #1359

updates `td` prose to include mention of how the `headers` attribute can be used to explicity indicate headers for a particular cell.

Revise note to refer to tables example to indicate that the `td`’s content attributes (colspan, rowspan, headers) will have examples in one/both of the linked sections.

Add table with `headers` example to table examples at end of file.

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -2950,4 +2954,77 @@
+    </table>
+  </xmp>
+
+  This table could look like this:
```
that wording makes sense when the example is illustrated with a picture of what it might produce. I think it makes more sense to say something like

> this example produces the following table (in your browser)

scottaohara replied:
@chaals i was following how the other table examples were noted.  I agree with you though, but I think I should go in and change all instances of this.

scottaohara replied:
Though, there is CSS that is applied to these tables, so *could* isn't completely inappropriate here, as by default these tables won't look like the output in the spec.

scottaohara replied:
@chaals I've updated the examples per your suggestion, but also added a bit of introductory text to the section so we only have to mention "in your browser" once, as well as allude to the fact there's some CSS applied to the tables for styling.

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -1376,12 +1376,16 @@
-  those header cells, if any, instead of the contents of the header cells themselves.
+  (as determined by the <a>algorithm for assigning header cells</a>).
+
+  When a cell's header cells are being listed, user agents may use the value of <code>abbr</code>
```
again, `<{th/abbr}>` is better.

chaals reviewed (dismissed):
couple of minor comments, but overall looks good.

scottaohara committed:
update abbr markup

change <code>abbr</code>
to
<{th/abbr}>

scottaohara committed:
update prose between markup & output

adds new intro text

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -1376,12 +1376,16 @@
-  those header cells, if any, instead of the contents of the header cells themselves.
+  (as determined by the <a>algorithm for assigning header cells</a>).
+
+  When a cell's header cells are being listed, user agents may use the value of <{th/abbr}>
```
Think  you should be clearer about what they may use it for. Something like

> may use...if any **e.g. to present to the user through a voice interface**

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -1376,12 +1376,16 @@
+
+  When a cell's header cells are being listed, user agents may use the value of <{th/abbr}>
+  attributes on those header cells, if any, instead of the contents of the header cells themselves.
+  A cell's header cell or cells are indicated by <{th}> elements, or by explicitly indicating a
```
mention and link to the bit that determines what headers apply where?

scottaohara commented:
@chaals any more that needed to be done on this?

chaals reviewed (commented):
Couple more trivial comments. Feel free to deal with them or not, and then merge

scottaohara committed:
addressing comments

reference how `abbr` might be used for header cell.

Add note to reference forming relationships between data cells and header cells.

chaals committed:
editing wording (#1487)

maybe smoother

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -1339,9 +1345,9 @@
-      A <{td}> element's <a>end tag</a> may be omitted if the <{td}> element is immediately
-      followed by a <code>td</code> or <{th}> element, or if there is no more content in the
-      parent element.
+      A <{td}> element's <a>end tag</a> may be omitted if the <code>td</code> element is
```
just remove the second "td"?

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -1339,9 +1345,9 @@
-      parent element.
+      A <{td}> element's <a>end tag</a> may be omitted if the <code>td</code> element is
+      immediately followed by a <code>td</code> or <{th}> element, or if there is no more
+      content in the parent element.
```
Does this mean "...followed by a `<{td}>`, `<{th}>`, or the closing `</tr>` tag or another tag that automatically closes its parent `<{tr}>` element?

chaals commented on sections/semantics-tabular-data.include:
```diff
@@ -2950,4 +2965,77 @@
+    </table>
+  </xmp>
+
+  This example produces the following table:
```
word as "...following table **in your browser**" ?

chaals reviewed (commented):
couple of questions / suggestions. Feel free to merge if you want to ignore them - otherwise it looks good already.

scottaohara committed:
some wording updates
